design_docs/analysis_task_template.md

```diff
@@ -0,0 +1,741 @@
+    endDate = '{:04d}-12-31_23:59:59'.format(
+        config.getint(section, 'endYear'))
+    if not config.has_option(section, 'endDate'):
+        config.set(section, 'endDate', endDate)
+
+    return  # }}}
+```
+Much of this code has been taken out of `run_analysis.py`, simplifying and clarifying
+the code.
+
+<h4> Implementation: Analysis Continues even when Analysis Task Fails <br>
+Date last modified: 2017/03/16 <br>
+Contributors: Xylar Asay-Davis
+</h4>
+
+Calls to `check` and `run` methods in `run_analysis.py` are inside of 
+`try/except` blocks, which catch the exceptions and print the stack trace 
+but don't cause the code to exit.
+```python
+try:
+    analysisTask.check()
+...
+except:
+    traceback.print_exc(file=sys.stdout)
+    print "ERROR: analysis module {} failed during check and " \
+        "will not be run".format(analysisTask.taskName)
+
+...
+
+try:
+    analysisModule.run()
+except:
+    traceback.print_exc(file=sys.stdout)
+    print "ERROR: analysis module {} failed during run".format(
+        analysisTask.taskName)
+```
+
+<h4> Implementation: List of Tasks to Perform<br>
+Date last modified: 2017/03/16 <br>
+Contributors: Xylar Asay-Davis
+</h4>
+
+The tasks are imported and added to an anaysis list as follows:
+```python
+analyses = []
+
+# Ocean Analyses
+from mpas_analysis.ocean.time_series_ohc import TimeSeriesOHC
+analyses.append(TimeSeriesOHC(config, streamMap=oceanStreamMap,
+                              variableMap=oceanVariableMap))
+from mpas_analysis.ocean.time_series_sst import TimeSeriesSST
+analyses.append(TimeSeriesSST(config, streamMap=oceanStreamMap,
+                              variableMap=oceanVariableMap))
+
+from mpas_analysis.ocean.climatology_map import ClimatologyMap \
+    as ClimatologyMapOcean
+for fieldName in ['sst', 'mld', 'sss']:
+    analyses.append(ClimatologyMapOcean(config, streamMap=oceanStreamMap,
+                                        variableMap=oceanVariableMap,
+                                        fieldName=fieldName))
+
+# Sea Ice Analyses
+from mpas_analysis.sea_ice.timeseries import TimeSeries as TimeSeriesSeaIce
+analyses.append(TimeSeriesSeaIce(config, streamMap=seaIceStreamMap,
+                                 variableMap=seaIceVariableMap))
+from mpas_analysis.sea_ice.climatology_map import ClimatologyMap \
+    as ClimatologyMapSeaIce
+analyses.append(ClimatologyMapSeaIce(config, streamMap=seaIceStreamMap,
+                                     variableMap=seaIceVariableMap))
+```
+The `analyses` list is a list of instances of subclasses of `AnalysisTask`.
+
+Subsequent calls to analysis functions can loop over analyses, as in the following
+example for calling `run`:
+```python
+# run each analysis task
+for analysisTask in analyses:
+    try:
+        analysisTask.run()
+    except:
+        traceback.print_exc(file=sys.stdout)
+        print "ERROR: analysis module {} failed during run".format(
+            analysisTask.taskName)
+```
```

Comment:
Maybe something like https://taskman.readthedocs.io/en/latest/ (presumably one should be out there already), but I was really thinking of the one you made.